sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java

```diff
@@ -0,0 +1,269 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import java.time.Duration;
+import java.util.List;
+
+/**
+ * Represents the connection config with {@link ConnectionMode#DIRECT} associated with Cosmos Client in the Azure Cosmos DB database service.
+ */
+public final class DirectConnectionConfig extends ConnectionConfig {
+    //  Constants
+    private static final Duration DEFAULT_IDLE_CHANNEL_TIMEOUT = Duration.ZERO;
+    private static final Duration DEFAULT_IDLE_ENDPOINT_TIMEOUT = Duration.ofSeconds(70L);
+    private static final int DEFAULT_MAX_CHANNELS_PER_ENDPOINT = 30;
+    private static final int DEFAULT_MAX_REQUESTS_PER_ENDPOINT = 10;
+
+    private static final DirectConnectionConfig defaultConfig = new DirectConnectionConfig();
+
+    private Duration connectionTimeout;
+    private Duration idleChannelTimeout;
+    private Duration idleEndpointTimeout;
+    private int maxChannelsPerEndpoint;
+    private int maxRequestsPerChannel;
+
+    /**
+     * Constructor.
+     */
+    public DirectConnectionConfig() {
+        super(ConnectionMode.DIRECT);
+        this.connectionTimeout = null;
+        this.idleChannelTimeout = DEFAULT_IDLE_CHANNEL_TIMEOUT;
+        this.idleEndpointTimeout = DEFAULT_IDLE_ENDPOINT_TIMEOUT;
+        this.maxChannelsPerEndpoint = DEFAULT_MAX_CHANNELS_PER_ENDPOINT;
+        this.maxRequestsPerChannel = DEFAULT_MAX_REQUESTS_PER_ENDPOINT;
+    }
+
+    /**
+     * Gets the default DIRECT connection configuration.
+     *
+     * @return the default direct connection configuration.
+     */
+    public static DirectConnectionConfig getDefaultConfig() {
+        return DirectConnectionConfig.defaultConfig;
+    }
+
+    //  TODO: (DANOBLE) - To update these docs.
+
+    /**
+     * Gets the direct connection timeout
+     * @return direct connection timeout
+     */
+    public Duration getConnectionTimeout() {
+        return connectionTimeout;
+    }
+
+    /**
+     *  Sets the direct connection timeout
+     * @param connectionTimeout the connection timeout
+     * @return the {@link DirectConnectionConfig}
+     */
+    public DirectConnectionConfig setConnectionTimeout(Duration connectionTimeout) {
+        this.connectionTimeout = connectionTimeout;
+        return this;
+    }
+
+    /**
+     * Gets the idle channel timeout
+     * @return idle channel timeout
+     */
+    public Duration getIdleChannelTimeout() {
```

Comment:
connection vs channel

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/10983